brave/brave-core
# Adblock regional filters selection

Fix brave/brave-browser#1931

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on

## Test Plan:

### Default Region

- Start with a clean profile
- Set locale to "fr"
  - *Windows*: This is done in `chrome://settings/languages`
  - *Linux*: Run brave via `LANGUAGE=fr_FR ./brave`
  - *Mac*: Use the Language & Region operating system settings as described in https://github.com/brave/brave-browser/issues/1238#issuecomment-426068531
- Restart Brave
- Visit `brave://adblock`
- Verify that French regional list is selected
- Test some French regional sites
- Uncheck the French regional list
- Restart Brave
- Visit `brave://adblock` and verify that the French regional list remains unchecked

### Regular Usage

- Visit `brave://adblock`
- Enable one or more regional lists by clicking the associated checkbox
- Visit `brave://components` and verify that the enabled lists show up as installed components
- Test some regional sites, here are the lists https://github.com/brave/ad-block/blob/master/lists/regions.h you can load a list URL (e.g., https://easylist-downloads.adblockplus.org/Liste_AR.txt) to see the resources which it blocks
- Disable one or more regional lists by unchecking the associated checkbox
- Restart Brave
- Visit `brave://components` and verify that the disabled lists no longer show up as installed components

## Reviewer Checklist:
